cmake/CxxQt.cmake

```diff
@@ -75,6 +90,13 @@ function(cxx_qt_import_crate)
     # See also the "Linking Object Libraries" and "Linking Object Libraries via $<TARGET_OBJECTS>" sections:
     # https://cmake.org/cmake/help/latest/command/target_link_libraries.html
     target_link_libraries(${CRATE} INTERFACE ${CRATE}_initializers $<TARGET_OBJECTS:${CRATE}_initializers>)
+
+    # Link the static library to Qt
+    # Note that we cannot do this on the final CRATE target as this is an interface
+    # which depends on the static library. Which then causes the link order to be
+    # incorrect with Qt before the static library, which then fails with ld.bfd
+    # https://stackoverflow.com/questions/51333069/how-do-the-library-selection-rules-differ-between-gold-and-the-standard-bfd-li
```

Comment:
```suggestion
    # which depends on the static library. If we do target_link_libraries on the ${CRATE} target,
    # the static library will not actually depend on the Qt modules, but be a kind of "sibling dependency", which CMake may reorder.
    # This can cause CMake to emit the wrong link order, with Qt before the static library, which then fails to build with ld.bfd
    # https://stackoverflow.com/questions/51333069/how-do-the-library-selection-rules-differ-between-gold-and-the-standard-bfd-li
```

Reply:
done